src/entities/schedulers/mempool-checker.js

```diff
@@ -0,0 +1,69 @@
+// @flow
+
+import _ from 'lodash'
+import type { Logger } from 'bunyan'
+import { helpers } from 'inversify-vanillajs-helpers'
+
+import type { RawDataProvider, Database } from '../../interfaces'
+import { TX_STATUS } from '../../blockchain/common'
+import SERVICE_IDENTIFIER from '../../constants/identifiers'
+import type { ShelleyTxType } from '../../blockchain/shelley/tx'
+import { sleep } from '../../utils'
+
+import BaseScheduler from './base-scheduler'
+
+class MempoolChecker extends BaseScheduler {
+  dataProvider: RawDataProvider
+
+  checkMempoolMillis: number
+
+  db: Database<ShelleyTxType>
+
+  constructor(
+    checkMempoolSeconds: number,
+    logger: Logger,
+    dataProvider: RawDataProvider,
+    db: Database<ShelleyTxType>,
+  ) {
+    super(logger)
+    this.checkMempoolMillis = checkMempoolSeconds * 1000
+    this.dataProvider = dataProvider
+    this.db = db
+
+    logger.debug('[MempoolChecker] Checking mempool every', checkMempoolSeconds, 'seconds')
+  }
+
+  async updateFailedTxs(txHashes: Array<string>): Promise<void> {
+    if (_.isEmpty(txHashes)) {
+      return
+    }
+    await this.db.updateTxsStatus(txHashes, TX_STATUS.TX_FAILED_STATUS)
+    for (const txHash of txHashes) {
+      await this.db.addNewTxToTransientSnapshots(txHash, TX_STATUS.TX_FAILED_STATUS)
+    }
+  }
```

Comment:
This function gotta be a bit more complex. We need to first check the current status for all txs and then ensure these conditions:

1. If tx is currently `Successful` - we DO NOT change status, this should be handled only thru a rollback, so tx first becomes pending, and then failed.

2. If tx is currently already failed - we do nothing and skip, because we must ensure the same failed tx is not unnecessary included in the transient snapshots multiple times (remember that failed transient set is non-accumulative like pending).

P.S.

`db.updateTxsStatus` must be fixed to also change the `last_update` field.